code/modules/hydroponics/compost_bin.dm

```diff
@@ -0,0 +1,167 @@
+	density = TRUE
+	anchored = TRUE
+	/// stage in the process, for timing purposes.
+	var/process_counter = 0
+	/// amount of biomass in the compost bin
+	var/biomass = 0
+	/// amount of compost in the compost bin
+	var/compost = 0
+	/// The maximum amount of biomass the compost bin can store.
+	var/biomass_capacity = BASE_BIOMASS_CAPACITY
+	/// The maximum amount of compost the compost bin can store.
+	var/compost_capacity = BASE_COMPOST_CAPACITY
+
+/obj/machinery/compost_bin/Initialize(mapload)
+	return ..()
+
+/obj/machinery/compost_bin/on_deconstruction()
+	// returns wood instead of the non-existent components
+	new /obj/item/stack/sheet/wood(loc, 10)
+	return ..()
+
+/obj/machinery/compost_bin/screwdriver_act(mob/living/user, obj/item/I)
+	// there are no screws either
+	to_chat(user, "<span class='warning'>[src] has no screws!</span>")
+	return TRUE
+
+/obj/machinery/compost_bin/crowbar_act(mob/living/user, obj/item/I)
+	// no panel either
+	return default_deconstruction_crowbar(user, I, ignore_panel = TRUE)
+
+// Accepts inserted plants and converts them to biomass
+/obj/machinery/compost_bin/proc/make_biomass(obj/item/reagent_containers/food/snacks/grown/O)
+	// calculate biomass from plant nutriment and plant matter
+	var/plant_biomass = O.reagents.get_reagent_amount("nutriment") + O.reagents.get_reagent_amount("plantmatter")
+	biomass += clamp(plant_biomass * 10, 1, biomass_capacity-biomass)
+	//plant delenda est
+	qdel(O)
+
+// takes care of plant insertion and conversion to biomass, and start composting what was inserted
+/obj/machinery/compost_bin/attackby(obj/item/O, mob/user, params)
+	if(user.a_intent == INTENT_HARM)
+		return ..()
+
+	if(istype(O, /obj/item/storage/bag/plants))
+		if(biomass >= biomass_capacity)
+			to_chat(user, "<span class='warning'>[src] can't hold any more biomass!</span>")
+			return
+
+		var/obj/item/storage/bag/plants/PB = O
+		for(var/obj/item/reagent_containers/food/snacks/grown/G in PB.contents)
+
+			if(biomass >= biomass_capacity)
+				break
+			PB.remove_from_storage(G, src)
+			make_biomass(G)
+
+		if(biomass < biomass_capacity)
+			to_chat(user, "<span class='info'>You empty [PB] into [src].</span>")
+		else
+			to_chat(user, "<span class='info'>You fill [src] to its capacity.</span>")
+
+		SStgui.update_uis(src)
+		update_icon_state()
+		return TRUE
+
+	if(istype(O, /obj/item/reagent_containers/food/snacks/grown))
+		if(biomass >= biomass_capacity)
+			to_chat(user, "<span class='warning'>[src] can't hold any more plants!</span>")
+			return
+		if(!user.unEquip(O))
+			return
+
+		O.forceMove(src)
+		make_biomass(O)
+		to_chat(user, "<span class='info'>You put [O] in [src].</span>")
+		SStgui.update_uis(src)
+		update_icon_state()
+		return TRUE
+
+	to_chat(user, "<span class='warning'>You cannot put this in [name]!</span>")
+
+//Compost compostable material if there is any
+/obj/machinery/compost_bin/process()
+	if(compost >= compost_capacity || biomass <= 0)
+		return
```

Comment:
```suggestion
	if(compost >= compost_capacity || biomass < MIN_CONVERSION)
		return
```
This allows you to skip a `min()` on line 107 as we are already at that amount, and it disallows printing compost with very low yield crops!